docs/generating-tests.rst

```diff
@@ -0,0 +1,417 @@
+                    "balance" : "2000000000000000010",
+                    "storage" : {
+                        "0x" : "0x01",
+                        "0x01" : "0x01"
+                    }
+                },
+                "2adc25665018aa1fe0e6bc666dac8fc2697ff9ba" : {
+                    "balance" : "20663"
+                },
+                "a94f5374fce5edbc8e2a8697c15331677e6ebf0b" : {
+                    "balance" : "99979327",
+                    "nonce" : "1"
+                }
+            }
+        },
+        {
+            "indexes" : {
+                "data" : 1,
+                "gas" : -1,
+                "value" : -1
+            },
+        ...
+        }
+    ]
+
+
+``indexes`` field specifies a subset of the transactions.  ``-1`` means "whichever".
+``"data" : 0`` points to the first element in the ``data`` field in ``transaction``.
+
+``network`` field is somehow similar.  It specifies the versions of the protocol
+for which the expectation applies.  For expectations common to all versions, say
+``"network" : ALL``.
+
+Filling the Test
+----------------
+
+The test filler file is not for consumption.  The filler file needs to be filled
+into a test.  ``testeth`` has the ability to compute the post-state from the test filler,
+and produce the test.  The advantage of the filled test is that it can catch any post-state
+difference between clients.
+
+First, if you created a new subdirectory for the filler, you need to edit the source
+of ``cpp-ethereum`` so that ``testeth`` recognizes the new subdirectory.  The file to 
+edit is `StateTests.cpp`_, which lists the names of the subdirectories scanned for 
+GeneralStateTest filters.
+
+.. _`StateTests.cpp`: https://github.com/ethereum/aleth/blob/master/test/tools/jsontests/StateTests.cpp
+
+
+After building ``testeth``, you are ready to fill the test.
+
+.. code:: bash
+
+   ETHEREUM_TEST_PATH="<LOCAL_PATH_TO_ETH_TESTS>" test/testeth -t GeneralStateTests/stExample2 -- --filltests --checkstate
+
+
+where the environmental variable ``ETHEREUM_TEST_PATH`` should point to the directory 
+where ``tests`` repository is checked out.  ``stExample2`` should be replaced with 
+the name of the subdirectory you are working on.  ``--filltests`` option tells ``testeth``
+to fill tests.  ``--checkstate`` tells ``testeth`` to check the final states against
+the ``expect`` fields.
+
+Depending on your shell, there are various ways to set up ``ETHEREUM_TEST_PATH``
+environment variable.  For example, adding ``export ETHEREUM_TEST_PATH=/path/to/tests``
+to ``~/.bashrc`` might work for ``bash`` users.
+
+``testeth`` with ``--filltests`` fills every test filler it finds. The command might
+modify existing test cases. After running ``testeth`` with ``--filltests``\ , try running
+``git status`` in the ``tests`` directory. If ``git status`` indicates changes in 
+unexpected files, that is an indication that the behavior of ``cpp-ethereum`` changed unexpectedly.
```

Comment:
Travis CI seems to be doing a check of the sourceHash in each filled file - it may be useful to explain what that command is as a way to identify which tests need to be refilled.

Reply:
I am not having a local up-to-date build of ``testeth`` up an running and so can't try it out myself, could you make a concrete suggestion for a ``singletest`` option addition text? Think that would be easiest.

Travis is using the [test.py](https://github.com/ethereum/tests/blob/develop/test.py) Python script to check, this is not part of the Aleth client (@winsvega: correct me if I am wrong). One should probably add to the docs that this Python tool exists in a addition also here and point to the ``test.py`` docs (currently in ``README``, should be moved to the docs as well though).

Reply:
test.py double check the test to be filled from latest fillers. when you run the `--filltests` command it will be checked for errors (expect section) automatically. 

you only use `--filltests` command when developing new test cases. if you just running the tests you must not use this command. `--singletest <TESTNAME> --singlenet <FORKNAME>` allow to select a specific test at specific fork. ` -d <DATAINDEX> -g <GASINDEX> -v <VALUEINDEX>` allow to select specific transaction from general state test.

Reply:
Have added these extra explanations for ``--singletest``.

Reply:
Maybe:

Alternatively, individual tests can be refilled using:
`ETHEREUM_TEST_PATH="<LOCAL_PATH_TO_ETH_TESTS>" test/testeth -t GeneralStateTests/stExample2 -- --filltests --singletest <test-name>`